### PR TITLE
[prometheus-exporters] Replace null objects with empty strings

### PR DESCRIPTION
**Pull request by witekest**
One of the old buggy versions of prometheus-exporters-formula was
serializing formula data with `null` objects as dictionary values. Such
formula data objects cannot be correctly serialized in Uyuni and produce
errors:

```
redstone.xmlrpc.XmlRpcDispatcher.writeError redstone.xmlrpc.XmlRpcException: ERROR IN SERIALIZER FOR com.redhat.rhn.domain.dto.FormulaData
```

To fix that we replace all `null` dictionary values with empty strings
in formula data JSON files.

### CODE REVIEW

File: prometheus-exporters-formula/scripts/migrate_formula_data.py [hunk 1]
```diff
@@ -53,8 +56,19 @@ def migrate(self):
...
-            with open(self.filepath, 'w') as output:
-                json.dump(self.data, output)
+
+    def dump(self):
+        with open(self.filepath, 'w') as output:
+            json.dump(self.data, output)
+
+
+def none_value_to_empty_string(data):
+    if type(data) is dict:
```
Comment (agraul) [inline prometheus-exporters-formula/scripts/migrate_formula_data.py:66]: Generally it is recommended to use `isinstance` for type checking and to use abstract types over concrete ones. The "pythonic" way would be to emprace duck-typing, e.g.

```python
try:
    for k, v in data.items():
        ...
except AttributeError:
    pass
```
If you want to check types, I would use
```python
    # at the top of scripts.migrate_formular_data
    from collections.abc import Mapping
    # here
    if isinstance(data, Mapping):
        ...
```

Comment (witekest) [reply prometheus-exporters-formula/scripts/migrate_formula_data.py]: Good point. Updated to check for the interface.

File: prometheus-exporters-formula/scripts/migrate_formula_data.py [hunk 2]
```diff
@@ -42,8 +44,11 @@ def migrate_from_version_1(self):
         for exporter in v1_keys:
             exporters[exporter] = self.data.pop(exporter)
 
+    def fix_schema(self):
+        none_value_to_empty_string(self.data)
+
     def migrate(self):
-        self.parse()
+
```
Comment (renner) [inline prometheus-exporters-formula/scripts/migrate_formula_data.py:51]:
```suggestion
```
Maybe we could remove this line completely?

**Review by renner — APPROVED** [review]
Looks good, no objections from my side!

**Pull request merged**